core/services/vrf/vrfcommon/inflight_cache.go

```diff
@@ -0,0 +1,82 @@
+package vrfcommon
+
+import (
+	"sync"
+
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+type InflightCache interface {
+	Add(lg types.Log)
+	Contains(lg types.Log) bool
+	Size() int
+}
+
+var _ InflightCache = (*inflightCache)(nil)
+
+const cachePruneInterval = 1000
```

Comment:
Should this be different for different chains?

Reply:
This is just the interval at which we see if we should prune stuff. Logs are only deleted past the chain's finality depth, which is chain specific.